src/cpp/database/database_queue.cpp

```diff
@@ -147,13 +159,16 @@ void DatabaseDataQueue::process_sample_type(
     sample.remote_locator = found_remote_locators.front().second;
 
     std::string guid = deserialize_guid(item.src_guid());
-    auto found_entities = database_->get_entities_by_guid(entity_kind, guid);
-    if (found_entities.empty())
+    try
+    {
+        auto found_entity = database_->get_entities_by_guid(entity_kind, guid);
+        domain = found_entity.first;
+        entity = found_entity.second;
+    }
+    catch (BadParameter& e)
```

Comment:
catch branch is not tested

Reply:
Because in the implementation the DataCount is always processed first, and the ByteCount afterwards. The DataCount processing fails on the same condition. I made a hack forcing to process both, no matter what, but I'm not convinced.